odoo/fields.py

```diff
@@ -778,28 +774,12 @@ def base_field(self):
     # Company-dependent fields
     #
 
-    def _default_company_dependent(self, model):
-        return model.env['ir.property']._get(self.name, self.model_name)
-
-    def _compute_company_dependent(self, records):
-        # read property as superuser, as the current user may not have access
-        Property = records.env['ir.property'].sudo()
-        values = Property._get_multi(self.name, self.model_name, records.ids)
-        for record in records:
-            record[self.name] = values.get(record.id)
-
-    def _inverse_company_dependent(self, records):
-        # update property as superuser, as the current user may not have access
-        Property = records.env['ir.property'].sudo()
-        values = {
-            record.id: self.convert_to_write(record[self.name], record)
-            for record in records
-        }
-        Property._set_multi(self.name, self.model_name, values)
-
-    def _search_company_dependent(self, records, operator, value):
-        Property = records.env['ir.property'].sudo()
-        return Property.search_multi(self.name, self.model_name, operator, value)
+    def get_company_dependent_fallback(self, records):
+        assert self.company_dependent
+        fallback = records.env['ir.default']._get_model_defaults(records._name).get(self.name)
+        fallback = self.convert_to_write(fallback, records)
+        fallback = self.convert_to_cache(fallback, records, validate=False)
```

Comment:
If I am not mistaken, the value you get from `ir.default` can directly be passed to `convert_to_cache`.
```suggestion
        fallback = records.env['ir.default']._get_model_defaults(records._name).get(self.name)
        fallback = self.convert_to_cache(fallback, records, validate=False)
```